aospy/calc.py

```diff
@@ -785,9 +808,19 @@ def _load_from_scratch(self, dtype_out_time, dtype_out_vert=False,
                            region=False):
         """Load aospy data saved on scratch file system."""
         ds = xr.open_dataset(self.path_scratch[dtype_out_time],
-                               engine='scipy')
+                             engine='scipy')
         if region:
-            return ds[region.name]
+            arr = ds[region.name]
+            # Use region-specific pressure values if available.
+            if self.dtype_in_vert == ETA_STR and not dtype_out_vert:
+                # try:
+                    arr[PFULL_STR].values = arr[region.name +
```

Comment:
>are the region-specific pressure coordinates and the regional variables themselves functions of the universal pressure coordinate PFULL_STR? 

Yes

>One way you could avoid this dropping maneuver would be to store the regional variables as functions of the region-specific pressure coordinates (and have the region-specific coordinates be functions of themselves (not the universal PFULL_STR)). Then in passing the region-specific DataArray to your plotting routines all you would need to do is select the region from the Dataset and rename its pressure coordinate to PFULL_STR. If things were structured this way, selecting from the Dataset would automatically take care of dropping the other coordinates. It would also remove the need to replace coordinate values.

I like that idea.

>It might be slightly cleaner down here, but perhaps upstream (in initially saving the data) it could complicate things, so you'd have to make that judgement.

Good point.  I think I will still ultimately implement your suggestion though -- I think it's better to have the data saved to disk with each region defined in terms of its own coordinates.  (Maybe also include the reference `pfull` array as an extra coord?)

FYI it will probably be a while (i.e. next week or later) before I'm able to make these changes...trying to finish a manuscript, and its sufficiently functional for now.